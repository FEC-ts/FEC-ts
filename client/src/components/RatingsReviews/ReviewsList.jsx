import React, { useState } from 'react';
import Review from './Review.jsx';
import MoreReviewsButton from './MoreReviewsButton.jsx';;
import SortMenu from './SortMenu.jsx';
import InnerReviewList from './InnerReviewList.jsx';

// John
// Remember sync and to GIT PULL

const ReviewsList = ({reviewInfo, setSortSelection, starArrayMaker}) => {

  const [reviewsDisplayed, setReviewsDisplayed] = useState(2);

  let currentlyDisplayedReviews = [];
  // console.log(reviewsDisplayed);
  const moreReviewsClickHandler = function () {
    setReviewsDisplayed(reviewsDisplayed + 2);
  }


  const whetherToDisplayMoreReviewsButton = function() {
    if (reviewInfo.length > 2
      && reviewsDisplayed < reviewInfo.length) {
        return (
          <div>
            <MoreReviewsButton
      moreReviewsClickHandler={moreReviewsClickHandler}/>
          </div>
        )
      }
  }

  if (reviewInfo) {
    currentlyDisplayedReviews = reviewInfo.slice(0, reviewsDisplayed);
  }

  if (reviewInfo) {
    return(
      <div className='ReviewsList'>
        <h2>Reviews List Section</h2>
        <SortMenu setSortSelection={setSortSelection}/>
        <InnerReviewList currentlyDisplayedReviews={currentlyDisplayedReviews} starArrayMaker={starArrayMaker}/>
        {whetherToDisplayMoreReviewsButton()}
      </div>
    )
  }
  return(
    <div>
      <progress></progress>
    </div>
  )
};

export default ReviewsList;