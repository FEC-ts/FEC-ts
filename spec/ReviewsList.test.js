/** @jest-environment jsdom */
import renderer from 'react-test-renderer';
import ReviewsList from '../client/src/components/RatingsReviews/ReviewsList.jsx';
import {cleanup, fireEvent, render, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

const testData = require('./ReviewStubs.js');
const user = userEvent.setup();

let starArrayMaker = function (starRating) {
  let fullStarCount = Math.floor(starRating);
  let emptyStarCount = 4 - fullStarCount;
  let partialStar = starRating - fullStarCount;
  let starArray = [];
  for (let i = 0; i < 5; i++) {
    if (i < fullStarCount) {
      starArray[i] = 1;
    } else if(i === fullStarCount) {
      starArray[i] = partialStar;
    } else {
      starArray[i] = 0;
    }
  }
  return starArray;
}

describe ('The Review List should having interactive elements', () => {
  it ('expect the reviews list to exist', () => {
    const component = renderer.create(
      <ReviewsList reviewInfo={testData.reviewInfo} reviewMetaData={testData.reviewMetaData} starArrayMaker={starArrayMaker}/>,
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it ('expect the reviews list to have a Sort on component', () => {
    const reviewList = render(<ReviewsList reviewInfo={testData.reviewInfo} reviewMetaData={testData.reviewMetaData} starArrayMaker={starArrayMaker}/>)
    expect(reviewList.queryByText('Sort on')).toBeDefined();
  })

  it ('expect the More Reviews button to appear when there are more than 2 reviews', () => {
    const reviewList = render(<ReviewsList reviewInfo={testData.reviewInfo} reviewMetaData={testData.reviewMetaData} starArrayMaker={starArrayMaker}/>)
    expect(reviewList.queryByRole('button', {name: 'More Reviews'})).toBeDefined();
  })

  it ('expect the More Reviews button to disappear when there are 2 or less reviews', () => {
    let shortList = testData.reviewInfo.slice(0, 2);
    const reviewList = render(<ReviewsList reviewInfo={shortList} reviewMetaData={testData.reviewMetaData} starArrayMaker={starArrayMaker}/>)
    reviewList.queryByText('More Reviews');
    expect(reviewList.queryByRole('button', {name: 'More Reviews'})).toBeNull();
  })

  it ('expect the More Reviews button to disappear when all reviews are displayed', () => {
    const reviewList = render(<ReviewsList reviewInfo={testData.reviewInfo} reviewMetaData={testData.reviewMetaData} starArrayMaker={starArrayMaker}/>)
    const MoreReviewsButton = reviewList.queryByRole('button', {name: 'More Reviews'});
    fireEvent.click(MoreReviewsButton)
    expect(reviewList.queryByRole('button', {name: 'More Reviews'})).toBeNull();
  })

  it ('expect the Add A Review button to be present with no reviews', () => {
    const reviewListNoReviews = render(<ReviewsList reviewInfo={[]} reviewMetaData={testData.reviewMetaData} starArrayMaker={starArrayMaker}/>)
    expect(reviewListNoReviews.queryByRole('button', {name: 'Add A Review'})).toBeDefined()
  })

  it ('expect the Add A Review button to be present with reviews', () => {
    const reviewList = render(<ReviewsList reviewInfo={testData.reviewInfo} reviewMetaData={testData.reviewMetaData} starArrayMaker={starArrayMaker}/>)
    expect(reviewList.queryByRole('button', {name: 'Add A Review'})).toBeDefined();
  })

  it ('expect the Add A Review button to make a new review form appear', () => {
    const reviewList = render(<ReviewsList reviewInfo={testData.reviewInfo} reviewMetaData={testData.reviewMetaData} starArrayMaker={starArrayMaker}
    productInfo={testData.productInfo}/>)
    const AddAReviewButton = reviewList.queryByRole('button', {name: 'Add A Review'});
    fireEvent.click(AddAReviewButton);
    expect(reviewList.queryByRole('form')).toBeDefined();
    expect(reviewList.queryByText('Write Your Review')).toBeDefined();
  })
})

describe ('The new review form should have required elements', () => {

  it ('Should have the required questions for the form', () => {
    const reviewList = render(<ReviewsList reviewInfo={testData.reviewInfo} reviewMetaData={testData.reviewMetaData} starArrayMaker={starArrayMaker}
      productInfo={testData.productInfo}/>)
    const AddAReviewButton = reviewList.queryByRole('button', {name: 'Add A Review'});
    fireEvent.click(AddAReviewButton);
    expect(reviewList.queryByText('Overall Rating *')).toBeDefined();
    expect(reviewList.queryByText('Do you recommend this product? *')).toBeDefined();
    expect(reviewList.queryByText('Characteristics *')).toBeDefined();
    expect(reviewList.queryByText('Review Summary')).toBeDefined();
    expect(reviewList.queryByText('Review Body *')).toBeDefined();
    expect(reviewList.queryByText('Upload Your Photos')).toBeDefined();
    expect(reviewList.queryByText('Your email')).toBeDefined();
    expect(reviewList.queryByText('Submit')).toBeDefined();
  })

  it ('Form should remain after submission mandatory fields are not filled', () => {
    const reviewList = render(<ReviewsList reviewInfo={testData.reviewInfo} reviewMetaData={testData.reviewMetaData} starArrayMaker={starArrayMaker}
      productInfo={testData.productInfo}/>)
    const AddAReviewButton = reviewList.queryByRole('button', {name: 'Add A Review'});
    fireEvent.click(AddAReviewButton);
    const RecommendInput = reviewList.queryByDisplayValue('Yes');
    fireEvent.click(RecommendInput);
    const characteristicsSet = reviewList.queryAllByDisplayValue(3);
    for (let i = 0; i < characteristicsSet.length; i += 1) {
      fireEvent.click(characteristicsSet[i]);
    }
    const submitButton = reviewList.queryByRole('button', {name: 'Submit Review'})
    fireEvent.click(submitButton);
    expect(reviewList.queryByRole('form')).toBeDefined();
    expect(reviewList.queryByText('Write Your Review')).toBeDefined();
  })

  it ('Form should disappear when the submission is valid', async () => {
    const reviewList = render(<ReviewsList reviewInfo={testData.reviewInfo} reviewMetaData={testData.reviewMetaData} starArrayMaker={starArrayMaker}
      productInfo={testData.productInfo}/>)
    const AddAReviewButton = reviewList.queryByRole('button', {name: 'Add A Review'});
    fireEvent.click(AddAReviewButton);
    const RatingStar = reviewList.queryByText('Overall Rating *')
      .querySelectorAll('div')[3];
    fireEvent.click(RatingStar);

    const RecommendInput = reviewList.queryByDisplayValue('Yes');
    fireEvent.click(RecommendInput);
    const characteristicsSet = reviewList.queryAllByDisplayValue(3);
    for (let i = 0; i < characteristicsSet.length; i += 1) {
      fireEvent.click(characteristicsSet[i]);
    }

    const ReviewBody = reviewList.queryByPlaceholderText('Why did you like the product or not?');

    await userEvent.type(ReviewBody, 'This is sample text meant to simulate a body with 50 characters.')

    const submitButton = reviewList.queryByRole('button', {name: 'Submit Review'})
    fireEvent.click(submitButton);

    expect(reviewList.queryByRole('button', {name: 'Submit Review'})).toBeNull();
  })

  it ('Form should be remain if non-Mandatory fields are filled', async () => {
    const reviewList = render(<ReviewsList reviewInfo={testData.reviewInfo} reviewMetaData={testData.reviewMetaData} starArrayMaker={starArrayMaker}
      productInfo={testData.productInfo}/>)
    const AddAReviewButton = reviewList.queryByRole('button', {name: 'Add A Review'});
    fireEvent.click(AddAReviewButton);

    const ReviewSummary = reviewList.queryByPlaceholderText('Example: Best purchase ever!');

    await userEvent.type(ReviewSummary, 'Excellent Sample purchase');

    const ReviewNickName = reviewList.queryByPlaceholderText('Example: jackson11!');

    await userEvent.type(ReviewNickName, 'Sample11');

    const ReviewEmail = reviewList.queryByPlaceholderText('Example: jackson11@email.com');

    await userEvent.type(ReviewEmail, 'Sample11@mailed.org');

    const submitButton = reviewList.queryByRole('button', {name: 'Submit Review'})
    fireEvent.click(submitButton);

    expect(reviewList.queryByRole('button', {name: 'Submit Review'})).toBeDefined();
  })
});

describe ('The report and helpful links to act when clicked', () => {
  it ('The helpful link to only disappear if server request is successful', async () => {
    const reviewList = render(<ReviewsList reviewInfo={testData.reviewInfo} reviewMetaData={testData.reviewMetaData} starArrayMaker={starArrayMaker}
      productInfo={testData.productInfo}/>)
    const HelpfulLink = reviewList.queryByText('Yes(29)');
    await userEvent.click(HelpfulLink);
    expect(reviewList.queryByText('Yes(29)')).toBeDefined();
  })

  it ('The helpful link to only disappear if server request is successful', async () => {
    const reviewList = render(<ReviewsList reviewInfo={testData.reviewInfo} reviewMetaData={testData.reviewMetaData} starArrayMaker={starArrayMaker}
      productInfo={testData.productInfo}/>)
    const ReportLink = reviewList.queryAllByText('Report')[0];
    await userEvent.click(ReportLink);
    expect(reviewList.queryAllByText('Report')[0]).toBeDefined();
  })
})
