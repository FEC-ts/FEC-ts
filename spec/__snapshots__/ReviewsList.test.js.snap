// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Review List should having interactive elements expect the reviews list to exist 1`] = `
<div
  className="ReviewsList"
>
  <div>
    <label>
      Sort on
      <select
        onChange={[Function]}
      >
        <option
          value="relevant"
        >
          relevant
        </option>
        <option
          value="helpful"
        >
          helpful
        </option>
        <option
          value="newest"
        >
          newest
        </option>
      </select>
    </label>
  </div>
  <div
    className="innerReviewList"
  >
    <div
      className="oneReview"
    >
      <div>
        <div
          className="oneReviewRatingStars"
        >
          <svg
            className="ReviewRatingStars"
            fill="none"
            viewBox="0 0 12 12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m5.28347 1.54595c.29345-.594594 1.14132-.594592 1.43478 0l1.11172 2.25261 2.48593.36123c.6562.09534.9182.90172.4433 1.36455l-1.79877 1.75341.42464 2.47586c.11209.65349-.57385 1.15189-1.16076.84339l-2.22345-1.16899-2.22346 1.16899c-.58691.3085-1.27285-.1899-1.16076-.84339l.42464-2.47586-1.79882-1.75341c-.474809-.46283-.2128-1.26921.44338-1.36455l2.4859-.36123z"
              fill="#212121"
            />
          </svg>
          <svg
            className="ReviewRatingStars"
            fill="none"
            viewBox="0 0 12 12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m5.28347 1.54595c.29345-.594594 1.14132-.594592 1.43478 0l1.11172 2.25261 2.48593.36123c.6562.09534.9182.90172.4433 1.36455l-1.79877 1.75341.42464 2.47586c.11209.65349-.57385 1.15189-1.16076.84339l-2.22345-1.16899-2.22346 1.16899c-.58691.3085-1.27285-.1899-1.16076-.84339l.42464-2.47586-1.79882-1.75341c-.474809-.46283-.2128-1.26921.44338-1.36455l2.4859-.36123zm.71739.80597-.97892 1.9835c-.11653.23611-.34178.39977-.60235.43763l-2.18892.31807 1.58392 1.54394c.18854.18379.27458.44859.23007.7081l-.37391 2.18008 1.95783-1.02929c.23306-.12253.51149-.12253.74455 0l1.95783 1.02929-.37391-2.18007c-.04451-.25952.04153-.52432.23008-.70811l1.58392-1.54394-2.18893-.31807c-.26056-.03786-.48582-.20152-.60235-.43763z"
              fill="#212121"
            />
          </svg>
          <svg
            className="ReviewRatingStars"
            fill="none"
            viewBox="0 0 12 12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m5.28347 1.54595c.29345-.594594 1.14132-.594592 1.43478 0l1.11172 2.25261 2.48593.36123c.6562.09534.9182.90172.4433 1.36455l-1.79877 1.75341.42464 2.47586c.11209.65349-.57385 1.15189-1.16076.84339l-2.22345-1.16899-2.22346 1.16899c-.58691.3085-1.27285-.1899-1.16076-.84339l.42464-2.47586-1.79882-1.75341c-.474809-.46283-.2128-1.26921.44338-1.36455l2.4859-.36123zm.71739.80597-.97892 1.9835c-.11653.23611-.34178.39977-.60235.43763l-2.18892.31807 1.58392 1.54394c.18854.18379.27458.44859.23007.7081l-.37391 2.18008 1.95783-1.02929c.23306-.12253.51149-.12253.74455 0l1.95783 1.02929-.37391-2.18007c-.04451-.25952.04153-.52432.23008-.70811l1.58392-1.54394-2.18893-.31807c-.26056-.03786-.48582-.20152-.60235-.43763z"
              fill="#212121"
            />
          </svg>
          <svg
            className="ReviewRatingStars"
            fill="none"
            viewBox="0 0 12 12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m5.28347 1.54595c.29345-.594594 1.14132-.594592 1.43478 0l1.11172 2.25261 2.48593.36123c.6562.09534.9182.90172.4433 1.36455l-1.79877 1.75341.42464 2.47586c.11209.65349-.57385 1.15189-1.16076.84339l-2.22345-1.16899-2.22346 1.16899c-.58691.3085-1.27285-.1899-1.16076-.84339l.42464-2.47586-1.79882-1.75341c-.474809-.46283-.2128-1.26921.44338-1.36455l2.4859-.36123zm.71739.80597-.97892 1.9835c-.11653.23611-.34178.39977-.60235.43763l-2.18892.31807 1.58392 1.54394c.18854.18379.27458.44859.23007.7081l-.37391 2.18008 1.95783-1.02929c.23306-.12253.51149-.12253.74455 0l1.95783 1.02929-.37391-2.18007c-.04451-.25952.04153-.52432.23008-.70811l1.58392-1.54394-2.18893-.31807c-.26056-.03786-.48582-.20152-.60235-.43763z"
              fill="#212121"
            />
          </svg>
          <svg
            className="ReviewRatingStars"
            fill="none"
            viewBox="0 0 12 12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m5.28347 1.54595c.29345-.594594 1.14132-.594592 1.43478 0l1.11172 2.25261 2.48593.36123c.6562.09534.9182.90172.4433 1.36455l-1.79877 1.75341.42464 2.47586c.11209.65349-.57385 1.15189-1.16076.84339l-2.22345-1.16899-2.22346 1.16899c-.58691.3085-1.27285-.1899-1.16076-.84339l.42464-2.47586-1.79882-1.75341c-.474809-.46283-.2128-1.26921.44338-1.36455l2.4859-.36123zm.71739.80597-.97892 1.9835c-.11653.23611-.34178.39977-.60235.43763l-2.18892.31807 1.58392 1.54394c.18854.18379.27458.44859.23007.7081l-.37391 2.18008 1.95783-1.02929c.23306-.12253.51149-.12253.74455 0l1.95783 1.02929-.37391-2.18007c-.04451-.25952.04153-.52432.23008-.70811l1.58392-1.54394-2.18893-.31807c-.26056-.03786-.48582-.20152-.60235-.43763z"
              fill="#212121"
            />
          </svg>
        </div>
        <div
          className="reviewerNameAndDate"
        >
          Roselyn_West11
          <span>
            , 
          </span>
          April 23, 2021
        </div>
        <div
          className="responseFromSeller"
        >
          <strong>
             Response from Seller 
          </strong>
          <div>
             
            "Expedita repellat quaerat debitis ab qui eos est voluptatibus."
          </div>
        </div>
      </div>
      <p>
        <strong>
          Mollitia nemo sunt.
        </strong>
      </p>
      <div
        className="oneReviewBody"
      >
        Another testing sample.
      </div>
      <div>
        <img
          className="oneReviewImage"
          onClick={[Function]}
        />
      </div>
      <div>
        Was this review helpful?
        <div
          className="wasItHelpful"
          onClick={[Function]}
        >
          Yes
          (
          29
          )
        </div>
        <span
          className="wasItHelpfulLine"
        >
           | 
        </span>
        <div
          className="wasItHelpful"
          onClick={[Function]}
        >
          Report
        </div>
      </div>
    </div>
    <div
      className="oneReview"
    >
      <div>
        <div
          className="oneReviewRatingStars"
        >
          <svg
            className="ReviewRatingStars"
            fill="none"
            viewBox="0 0 12 12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m5.28347 1.54595c.29345-.594594 1.14132-.594592 1.43478 0l1.11172 2.25261 2.48593.36123c.6562.09534.9182.90172.4433 1.36455l-1.79877 1.75341.42464 2.47586c.11209.65349-.57385 1.15189-1.16076.84339l-2.22345-1.16899-2.22346 1.16899c-.58691.3085-1.27285-.1899-1.16076-.84339l.42464-2.47586-1.79882-1.75341c-.474809-.46283-.2128-1.26921.44338-1.36455l2.4859-.36123z"
              fill="#212121"
            />
          </svg>
          <svg
            className="ReviewRatingStars"
            fill="none"
            viewBox="0 0 12 12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m5.28347 1.54595c.29345-.594594 1.14132-.594592 1.43478 0l1.11172 2.25261 2.48593.36123c.6562.09534.9182.90172.4433 1.36455l-1.79877 1.75341.42464 2.47586c.11209.65349-.57385 1.15189-1.16076.84339l-2.22345-1.16899-2.22346 1.16899c-.58691.3085-1.27285-.1899-1.16076-.84339l.42464-2.47586-1.79882-1.75341c-.474809-.46283-.2128-1.26921.44338-1.36455l2.4859-.36123zm.71739.80597-.97892 1.9835c-.11653.23611-.34178.39977-.60235.43763l-2.18892.31807 1.58392 1.54394c.18854.18379.27458.44859.23007.7081l-.37391 2.18008 1.95783-1.02929c.23306-.12253.51149-.12253.74455 0l1.95783 1.02929-.37391-2.18007c-.04451-.25952.04153-.52432.23008-.70811l1.58392-1.54394-2.18893-.31807c-.26056-.03786-.48582-.20152-.60235-.43763z"
              fill="#212121"
            />
          </svg>
          <svg
            className="ReviewRatingStars"
            fill="none"
            viewBox="0 0 12 12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m5.28347 1.54595c.29345-.594594 1.14132-.594592 1.43478 0l1.11172 2.25261 2.48593.36123c.6562.09534.9182.90172.4433 1.36455l-1.79877 1.75341.42464 2.47586c.11209.65349-.57385 1.15189-1.16076.84339l-2.22345-1.16899-2.22346 1.16899c-.58691.3085-1.27285-.1899-1.16076-.84339l.42464-2.47586-1.79882-1.75341c-.474809-.46283-.2128-1.26921.44338-1.36455l2.4859-.36123zm.71739.80597-.97892 1.9835c-.11653.23611-.34178.39977-.60235.43763l-2.18892.31807 1.58392 1.54394c.18854.18379.27458.44859.23007.7081l-.37391 2.18008 1.95783-1.02929c.23306-.12253.51149-.12253.74455 0l1.95783 1.02929-.37391-2.18007c-.04451-.25952.04153-.52432.23008-.70811l1.58392-1.54394-2.18893-.31807c-.26056-.03786-.48582-.20152-.60235-.43763z"
              fill="#212121"
            />
          </svg>
          <svg
            className="ReviewRatingStars"
            fill="none"
            viewBox="0 0 12 12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m5.28347 1.54595c.29345-.594594 1.14132-.594592 1.43478 0l1.11172 2.25261 2.48593.36123c.6562.09534.9182.90172.4433 1.36455l-1.79877 1.75341.42464 2.47586c.11209.65349-.57385 1.15189-1.16076.84339l-2.22345-1.16899-2.22346 1.16899c-.58691.3085-1.27285-.1899-1.16076-.84339l.42464-2.47586-1.79882-1.75341c-.474809-.46283-.2128-1.26921.44338-1.36455l2.4859-.36123zm.71739.80597-.97892 1.9835c-.11653.23611-.34178.39977-.60235.43763l-2.18892.31807 1.58392 1.54394c.18854.18379.27458.44859.23007.7081l-.37391 2.18008 1.95783-1.02929c.23306-.12253.51149-.12253.74455 0l1.95783 1.02929-.37391-2.18007c-.04451-.25952.04153-.52432.23008-.70811l1.58392-1.54394-2.18893-.31807c-.26056-.03786-.48582-.20152-.60235-.43763z"
              fill="#212121"
            />
          </svg>
          <svg
            className="ReviewRatingStars"
            fill="none"
            viewBox="0 0 12 12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m5.28347 1.54595c.29345-.594594 1.14132-.594592 1.43478 0l1.11172 2.25261 2.48593.36123c.6562.09534.9182.90172.4433 1.36455l-1.79877 1.75341.42464 2.47586c.11209.65349-.57385 1.15189-1.16076.84339l-2.22345-1.16899-2.22346 1.16899c-.58691.3085-1.27285-.1899-1.16076-.84339l.42464-2.47586-1.79882-1.75341c-.474809-.46283-.2128-1.26921.44338-1.36455l2.4859-.36123zm.71739.80597-.97892 1.9835c-.11653.23611-.34178.39977-.60235.43763l-2.18892.31807 1.58392 1.54394c.18854.18379.27458.44859.23007.7081l-.37391 2.18008 1.95783-1.02929c.23306-.12253.51149-.12253.74455 0l1.95783 1.02929-.37391-2.18007c-.04451-.25952.04153-.52432.23008-.70811l1.58392-1.54394-2.18893-.31807c-.26056-.03786-.48582-.20152-.60235-.43763z"
              fill="#212121"
            />
          </svg>
        </div>
        <div
          className="reviewerNameAndDate"
        >
          Joseph_West11
          <span>
            , 
          </span>
          April 20, 2021
        </div>
        <div
          className="responseFromSeller"
        >
          <strong>
             Response from Seller 
          </strong>
          <div>
             
            "Expedita repellat quaerat debitis ab qui eos est voluptatibus."
          </div>
        </div>
      </div>
      <p>
        <strong>
          Mollitia nemo sunt.
        </strong>
      </p>
      <div
        className="oneReviewBody"
      >
        Quis maiores et pariatur et reprehenderit quos neque distinctio et. Qui quo quo laborum eaque pariatur. Nostrum natus non ut fuga dolorem est corporis. Eum quisquam veritatis aspernatur voluptatum officia sed dolor rerum. Aut dolores natus ducimus to...
        <div
          className="linkToClick"
          onClick={[Function]}
        >
          Show More
        </div>
      </div>
      <div>
        <img
          className="oneReviewImage"
          onClick={[Function]}
        />
      </div>
      <div>
        Was this review helpful?
        <div
          className="wasItHelpful"
          onClick={[Function]}
        >
          Yes
          (
          29
          )
        </div>
        <span
          className="wasItHelpfulLine"
        >
           | 
        </span>
        <div
          className="wasItHelpful"
          onClick={[Function]}
        >
          Report
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="moreReviewsButton"
    >
      <div>
        <button
          className="more-questions-button"
          onClick={[Function]}
        >
          More Reviews
        </button>
      </div>
    </div>
    <div
      className="moreReviewsButton"
    >
      <button
        className="more-questions-button"
        onClick={[Function]}
      >
        Add A Review
      </button>
    </div>
  </div>
</div>
`;
